python-package/lightgbm/compat.py

```diff
@@ -6,6 +6,7 @@
     from pandas import DataFrame as pd_DataFrame
     from pandas import Series as pd_Series
     from pandas import concat
+    from pandas.api.types import CategoricalDtype as pd_CategoricalDtype
```

Comment:
This is fine that this import works with the latest version. But will it make sense to try import like `from pandas import CategoricalDtype as pd_CategoricalDtype` in case they'll change internal structure of modules?
Refer to
https://github.com/microsoft/LightGBM/blob/99e0a4bd7b9e7c557e593ff9172799822abc4b7d/python-package/lightgbm/compat.py#L68-L73
WDYT?

Reply:
Yes, this makes sense, thank you. Not sure, but it feels more likely that they are going to change `pandas.api.types` than the top-level import of their types `from pandas import CategoricalDtype`. Especially, given that their current top-level init goes into `pandas.core.api` for CategoricalDtype.